pkg/kubectl/cmd/context.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"io"
+)
+
+type CmdContext struct {
+	In            io.Reader
+	Out           io.Writer
+	ErrOut        io.Writer
+	BaseCmd       string
+	CmdName       string
+	FullName      string
+	CmdDetailsURL string
+	SubCmdNames   []string
```

Comment:
I wasn't expecting to see this.  Can you try to plumb it through without this one?

The rest matches my expectations.